AbletonMCP_Remote_Script/handlers/clips.py

```
@@ -0,0 +1,810 @@
+
+    Args:
+        launch_mode: 0=trigger, 1=gate, 2=toggle, 3=repeat
+    """
+    try:
+        clip = _get_clip(song, track_index, clip_index)
+        launch_mode = int(launch_mode)
+        if launch_mode < 0 or launch_mode > 3:
+            msg = "launch_mode must be 0-3 (trigger/gate/toggle/repeat), got {0}".format(launch_mode)
+            if ctrl:
+                ctrl.log_message(msg)
+            raise ValueError(msg)
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)